cmds/report.c

```diff
@@ -820,11 +994,18 @@ static void report_diff(struct uftrace_data *handle, struct opts *opts)
 	pr_out("#  [%d] base: %s\t(from %s)\n", 0, handle->dirname, handle->info.cmdline);
 	pr_out("#  [%d] diff: %s\t(from %s)\n", 1, opts->diff, data.handle.info.cmdline);
 	pr_out("#\n");
-	pr_out(formats[f_idx], headers[h_idx][0], headers[h_idx][1], headers[h_idx][2],
-	       maxlen, "Function");
-	pr_out(formats[f_idx], line, line, line, maxlen, line);
 
-	print_and_delete(&diff_tree, true, NULL, print_function_diff);
+	if (diff_policy.full) {
+		setup_field(&output_fields, opts, &setup_default_field_diff_full,
+			field_diff_full_table, ARRAY_SIZE(field_diff_full_table));
+	} else {
+		setup_field(&output_fields, opts, &setup_default_field_diff,
+			field_diff_table, ARRAY_SIZE(field_diff_table));
+	}
+
+	print_header(&output_fields, "  ", "Function", 2);
+	print_line(&output_fields, 2);
+	print_and_delete(&diff_tree, true, NULL, print_function);
```

Comment:
It'd be nice if we could 3 whitespaces between fields when `diff_policy.full` is used.  It'd have too much info so we'd better considering readability.